### PR TITLE
refactor: move code to linkedin/datahub-gma.

### PR DESCRIPTION
**Pull request by jplaisted**
"GMA" (Generalized Metadata Architecture) is the backend of DataHub, and has been moved to its own repository.

This deletes the code that was moved and uses jars that GMA publishes to bintray to load it.

Note that not all of GMA was moved, but most of it. We may still move more things to the other repository in the future.

Fixes https://github.com/linkedin/datahub/issues/1911

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -26,6 +26,8 @@ project.ext.spec = [
     ]
 ]
 
+def GMA_VERSION = "0.0.1"
```
Comment (mars-lan) [inline build.gradle:29]: Maybe move this to `buildscript` section like `ext.pegasusVersion`?

File: gms/client/build.gradle [hunk 2]
```diff
@@ -7,9 +7,9 @@ dependencies {
 
   dataModel project(':li-utils')
 
-  compile project(':metadata-dao')
+  compile externalDependency.gmaDaoApi
```
Comment (mars-lan) [inline gms/client/build.gradle:10]: Let's group `externalDependency` together?

File: gms/impl/build.gradle [hunk 3]
```diff
@@ -6,14 +6,15 @@ dependencies {
   compile project(path: ':gms:api', configuration: 'dataTemplate')
   compile project(':li-utils')
   compile project(':metadata-models')
-  compile project(':metadata-dao')
-  compile project(path: ':metadata-dao', configuration: 'dataTemplate')
-  compile project(':metadata-dao-impl:ebean-dao')
-  compile project(':metadata-dao-impl:elasticsearch-dao')
-  compile project(':metadata-dao-impl:neo4j-dao')
+  compile externalDependency.gmaDaoApi
```
Comment (mars-lan) [inline gms/impl/build.gradle:9]: ditto

File: metadata-dao-impl/kafka-producer/build.gradle [hunk 4]
```diff
@@ -1,7 +1,7 @@
 apply plugin: 'java'
 
 dependencies {
-  compile project(':metadata-dao')
+  compile externalDependency.gmaDaoApi
```
Comment (mars-lan) [inline metadata-dao-impl/kafka-producer/build.gradle:4]: ditto

File: metadata-dao-impl/restli-dao/build.gradle [hunk 5]
```diff
@@ -2,16 +2,19 @@ apply plugin: 'java'
 apply plugin: 'pegasus'
 
 dependencies {
-  compile project(':metadata-dao')
+  compile externalDependency.gmaDaoApi
```
Comment (mars-lan) [inline metadata-dao-impl/restli-dao/build.gradle:5]: ditto

**Review by mars-lan — CHANGES_REQUESTED** [review]

File: gms/client/build.gradle (hunk 2 above)
Comment (jplaisted) [reply gms/client/build.gradle:10]: fyi this was done with `sed`, which is why things stayed on the same line.

File: gms/api/build.gradle [hunk 6]
```diff
@@ -1,9 +1,9 @@
 apply plugin: 'pegasus'
 
 dependencies {
-  dataModel project(':metadata-dao')
+  dataModel externalDependency.gmaDaoApiDataTemplate
```
Comment (mars-lan) [inline gms/api/build.gradle:4]: Reorder?

File: metadata-dao-impl/restli-dao/build.gradle [hunk 7]
```diff
@@ -2,16 +2,19 @@ apply plugin: 'java'
 apply plugin: 'pegasus'
 
 dependencies {
-  compile project(':metadata-dao')
   compile project(':metadata-models')
-  compile project(':metadata-restli-resource')
+  compile externalDependency.gmaDaoApi
+  compile externalDependency.gmaDaoApiDataTemplate
```
Comment (mars-lan) [inline metadata-dao-impl/restli-dao/build.gradle:7]: Order alphabetically?

File: metadata-dao-impl/restli-dao/build.gradle [hunk 8]
```diff
@@ -2,16 +2,19 @@ apply plugin: 'java'
...
+  compile externalDependency.gmaRestliResources
+  compile externalDependency.gmaRestliResourcesDataTemplate
   compile externalDependency.cacheApi
   compile spec.product.pegasus.restliClient
   compile spec.product.pegasus.restliCommon
 
   compileOnly externalDependency.lombok
 
-  testCompile project(':metadata-testing:metadata-test-models')
+  testCompile externalDependency.gmaTestModels
```
Comment (mars-lan) [inline metadata-dao-impl/restli-dao/build.gradle:16]: reorder?

File: metadata-jobs/mae-consumer-job/build.gradle [hunk 9]
```diff
@@ -22,8 +22,8 @@ dependencies {
     compile project(':metadata-events:mxe-avro-1.7')
     compile project(':metadata-events:mxe-registration')
     compile project(':metadata-events:mxe-utils-avro-1.7')
-    compile project(':metadata-dao')
-    compile project(':metadata-dao-impl:neo4j-dao')
+    compile externalDependency.gmaDaoApi
```
Comment (mars-lan) [inline metadata-jobs/mae-consumer-job/build.gradle:25]: alphabetical order?

File: metadata-utils/build.gradle [hunk 10]
```diff
@@ -11,7 +11,7 @@ dependencies {
   compile spec.product.pegasus.restliCommon
 
   compile project(':li-utils')
-  compile project(':metadata-dao')
+  compile externalDependency.gmaDaoApi
```
Comment (mars-lan) [inline metadata-utils/build.gradle:14]: reorder?

**Review by mars-lan — CHANGES_REQUESTED** [review]

**Review by mars-lan — APPROVED** [review]
Great work!

**Pull request merged**